app/interactors/validate_user.rb

```diff
@@ -0,0 +1,11 @@
+class ValidateUser
+  include Interactor
+
+  def call
+    context.user = User.find_by(email: context.email.downcase)
+    user = context.user
+    password = context.password
+    errors = context.user.errors
+    context.fail!(errors: errors) unless user && user.authenticate(password)
```

Comment:
@npauzenga: there're a couple of ways you can go here. You could create a new private method that does this:

```ruby
def authenticate_user(user, password)
  return if user && user.authenticate(password)
  context.fail!(errors: user.errors)
end
```

Then your `call` method would look like this:

```ruby
def call
  context.user = User.find_by(email: context.email.downcase)
  authenticate_user(context.user, context.password)
end
```

You can also wrap `fail!`` under the `unless` call (you don't need to always force the single line `unless`/`if` :smiley:). It would look something like:

```ruby
def call
  context.user = User.find_by(email: context.email.downcase)
  unless context.user && context.user.authenticate(context.password)
    context.fail!(errors: context.user.errors)
  end
end
```

Either way is fine, it just depends on what you like best :smiley:. I would definitely prefer either one over creating extra "holder" variables :stuck_out_tongue:.